File: sorting/insertion-sort/insertion_sort.cpp
#include <bits/stdc++.h>
using namespace std;
int main()
{
    int n;
    cout<<"enter number of elements: ";
    cin>>n;
    int arr[n];

    // insertion sort logic
    int key, j;
    for(int i = 1; i < n; i++)
    {
        key = arr[i];
        j = i - 1;
        while (j >= 0 && arr[j] > key)
        {
            arr[j + 1] = arr[j];
            j = j - 1;
        }
        arr[j + 1] = key;
    }

    // print sorted element
    for(int i = 0; i < n; i++)
        cout << arr[i] << " ";
    cout << endl;

    return 0;
}

File: sorting/insertion-sort/insertion_sort.py
# Python program for implementation of Insertion Sort 

# Function to do insertion sort 
def insertionSort(arr):  
	for i in range(1, len(arr)): 

		key = arr[i] 
		j = i-1
		while j >=0 and key < arr[j] : 
				arr[j+1] = arr[j] 
				j -= 1
		arr[j+1] = key 


# Driver code to test above
n=int(input("enter lenght of array: "))
for i in range(0,n):
    arr[i]=int(input()) 
insertionSort(arr) 
print("Sorted array is:") 
for i in range(len(arr)): 
	print(arr[i])

